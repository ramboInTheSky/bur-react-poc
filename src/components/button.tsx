import React, { FC } from 'react';
import styled from 'styled-components';

const Container = styled.div``;
const Btn = styled.button``;

interface Props {
    readonly label: string;
    readonly disabled?: boolean;
    readonly onClick?: () => void;
    readonly type?: "button" | "reset" | "submit";
}

export const Button: FC<Props> = ({ disabled = false, label, ...props }) => 
<Container>
    <Btn {...props} disabled={disabled}>{label}</Btn>
</Container>;