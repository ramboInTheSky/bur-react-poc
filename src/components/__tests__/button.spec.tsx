import React from 'react';
import { shallow } from 'enzyme';

import { Button } from '../button';

describe(Button, () => {
  const props = {
    label: 'hello',
  };

  it('should render correctly', () => {
    expect(shallow(<Button {...props} />)).toMatchSnapshot();
  });

  it('should render correctly when disabled', () => {
    expect(shallow(<Button {...props} disabled />)).toMatchSnapshot();
  });
});
