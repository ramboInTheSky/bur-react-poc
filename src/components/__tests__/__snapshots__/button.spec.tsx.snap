// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<styled.div>
  <styled.button
    disabled={false}
  >
    hello
  </styled.button>
</styled.div>
`;

exports[`Button should render correctly when disabled 1`] = `
<styled.div>
  <styled.button
    disabled={true}
  >
    hello
  </styled.button>
</styled.div>
`;
